pkg/cloudprovider/vsphere/loadbalancer/README.md

```diff
@@ -0,0 +1,145 @@
+
+*Kubernetes load balancer support using NSX-T for the `vSphere` cloud controller
+manager.*
+
+This package enriches the cloud provider interface by implementing the load
+balancing API of the cloud controller for an NSX-T environment.
+
+The basic assumption is that all nodes are bound to a logical tier1 router.
+Here the load balancer service is attached to. Because there may be only
+one such service here, the configuration of the service must be done
+during the creation of the service. Here the selection of the (t-shirt) size is
+required (S, M or L)  
+For every service port a dedicated virtual server is managed which is
+connected to this load balancer service.
+
+## Features
+
+The load balancer controller part of the vsphere cloud controller manager
+is optional. If no `LoadBalancer` or `LoadBalancerClass` section is given
+in the controller configuration loadbalancing support is disabled.
+If enabled the follwing feature are supported:
+
+### Tagging
+
+All generated NSX-T elements are tagged with the app name of the controller,
+the cluster name and the information from the service. Using this tagging it is
+able to handle recovery of lost or dangling elements and garbage collection of
+unused elements previously generated by the controller, even if the kubernetes
+service object is already (accidentally) gone.
+
+### Load Balancer Classes
+
+This load balancer controller supports the usage of multiple load balancer
+classes. Classes are preconfigured in the configuration file of the cloud
+controller manager. There may be an arbitrary set of such classes in a dedicated
+setup. Every class may use another `IPPool` configured in NSX-T.
+This supports the creation of load balancers in different visibility realms,
+for example an `*internet facing* or a *private* load balancer. The IPPools must
+be preconfigured in NSX-T.
+Additionally dedicated TCP and/or UDP profiles can be selected differing from
+the default ones.
+
+The class used to create a Kubernetes load balancer can then be selected on
+the level of the Kubernetes service object.
+To select a dedicated load balancer class different from the default one, the
+Kubernetes service object must be annotated with the annotation:
+
+```yaml
+loadbalancer.vmware.io/class: <class name>
+```
+
+If no such annotation is given the default class will be used. This gives
+the adminstrator of the cluster a chance to restrict the usage of the
+NSXT-T resources for cluster users. He can determine which elements should
+be used for a dedicated purpose. The cluster user just needs to know and select
+the purpose by annotating the appropriate load balancer class.
+
+### Health Checks
+
+For TCP load balancers a health check will be generated.
+
+## Configuration File
+
+The controller manager requires dedicated entries in the cloud controller's
+configuration file:
+
+```ini
+[LoadBalancer]
+ipPoolName = pool1
+#ipPoolId = 0815
+#tier1GatewayPath = /infra/tier-1s/12345
```

Comment:
Can you get rid of the commented fields and create a separate example actually using those fields?

Reply:
Instead I've added a reference section listing all possible attributes with their meaning and combinations.